### PR TITLE
feat: distinguish between global and local transporter

### PR DESCRIPTION
**Pull request by welkeyever**
#### What type of PR is this?
feat

#### Check the PR title.
- [x] This PR title match the format: \<type\>(optional scope): \<description\>
- [x] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
server 区分局部和全局 transporter

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: change transporter of the hertz instance with option will not effect global transporter, vice versa. Avoid potential problems affecting other instances by modifying the transporter.
zh(optional): 修改 hertz 实例的 transporter 不会影响全局默认的 transporter，反之亦然；避免潜在的因为修改 transporter 导致影响到其他实例的问题。

#### Which issue(s) this PR fixes:
None

### CODE REVIEW

File: pkg/route/engine.go [hunk 1]
```diff
@@ -199,14 +199,34 @@ func (engine *Engine) GetOptions() *config.Options {
...
+	return tName
+}
+
+// Deprecated: This only get the global default transporter - may not be the real one used by the engine.
+// Use engine.GetTransporterName for the real transporter used.
 func GetTransporterName() (tName string) {
 	defer func() {
 		err := recover()
-		if err != nil {
+		if err != nil || tName == "" {
```
Comment (li-jin-gou) [inline pkg/route/engine.go:229]: Abstract a constant

**Review by li-jin-gou — CHANGES_REQUESTED** [review]

File: pkg/route/engine.go (hunk 1 above)
Comment (welkeyever) [reply pkg/route/engine.go:231]: done

File: pkg/common/config/option.go [hunk 2]
```diff
@@ -68,6 +69,8 @@ type Options struct {
 	TraceLevel                   interface{}
 	ListenConfig                 *net.ListenConfig
 
+	TransporterNewer func(opt *Options) network.Transporter
```
Comment (Duslia) [inline pkg/common/config/option.go:72]: What does this name mean?

Comment (welkeyever) [reply pkg/common/config/option.go:72]: function interface of 'creating a transporter'

**Review by li-jin-gou — APPROVED** [review]

File: pkg/common/config/option.go [hunk 3]
```diff
@@ -68,11 +69,15 @@ type Options struct {
...
 	ListenConfig                 *net.ListenConfig
 
+	// TransporterNewer is the function to create a transporter.
+	TransporterNewer func(opt *Options) network.Transporter
 	// Registry is used for service registry.
+
 	Registry registry.Registry
 	// RegistryInfo is base info used for service registry.
 	RegistryInfo *registry.Info
 	// Enable automatically HTML template reloading mechanism.
```
Comment (FGYFFFF) [inline pkg/common/config/option.go:79]: 为啥加空格

**Review by FGYFFFF — APPROVED** [review]

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**